api/client.js

```diff
@@ -464,5 +469,31 @@ function uncapitalize(string) {
   return string.charAt(0).toLowerCase() + string.slice(1);
 }
 
+/**
+  Create an async handler which will either return a value to a supplied
+  callback, or call the appropriate method on the promise resolve/reject.
+  @param {Function} [callback] The optional callback.
+  @param {Function} [resolve] The optional promise resolve function.
+  @param {Function} [reject] The optional promise reject function.
+  @return {Function} The returned function takes two arguments (err, value).
+    If the the callback is a function the two arguments will be passed through
+    to the callback in the same order. If no callback is supplied, the promise
+    resolve or reject method will be called depending on the existence of an
+    error value.
+*/
+function createAsyncHandler (callback, resolve, reject) {
+  return (error, value) => {
+    if (typeof callback === 'function') {
```

Comment:
Just `if(callback)`?, so that we let it raise an error if a callback is passed and it's not a function?